jt-mosaic/src/main/java/it/geosolutions/jaiext/mosaic/MosaicOpImage.java

```diff
@@ -337,19 +334,46 @@ private static ImageLayout getTargetSampleColorModel(List sources, Range[] noDat
             result.setSampleModel(sm);
         } else {
             // take a leap of faith and assume we can manage this with a RGB output
-            setRGBLayout(result, first);
+            setRGBLayout(result, first, hasAlpha(sources));
         }
 
         return result;
     }
 
-    private static void setRGBLayout(ImageLayout result, RenderedImage reference) {
+    private static boolean hasAlpha(List sources) {
+        int numSources = sources.size();
+        boolean hasAlphaBand = false;
+        for (int i = 1; i < numSources; i++) {
+            RenderedImage source = (RenderedImage) sources.get(i);
+            SampleModel sourceSampleModel = source.getSampleModel();
+            ColorModel sourceColorModel = source.getColorModel();
+            hasAlphaBand |=
+                    sourceSampleModel.getNumBands() > 1
+                            && sourceColorModel.hasAlpha()
+                            && sourceColorModel.getTransparency() == ColorModel.TRANSLUCENT;
+        }
+        return hasAlphaBand;
+    }
+
+    private static void setRGBLayout(
+            ImageLayout result, RenderedImage reference, boolean hasAlpha) {
         ColorSpace cs = ColorSpace.getInstance(ColorSpace.CS_sRGB);
-        int[] nBits = { 8, 8, 8 };
-        ColorModel cm = new ComponentColorModel(cs, nBits, false, false, Transparency.OPAQUE,
-                DataBuffer.TYPE_BYTE);
-        SampleModel sm = cm.createCompatibleSampleModel(reference.getSampleModel().getWidth(),
-                reference.getSampleModel().getWidth());
+        ColorModel cm;
+        if (hasAlpha) {
+            int[] nBits = new int[] {8, 8, 8, 8};
+            cm =
+                    new ComponentColorModel(
+                            cs, nBits, true, false, Transparency.TRANSLUCENT, DataBuffer.TYPE_BYTE);
+        } else {
+            int[] nBits = new int[] {8, 8, 8};
+            cm =
+                    new ComponentColorModel(
+                            cs, nBits, false, false, Transparency.OPAQUE, DataBuffer.TYPE_BYTE);
+        }
+        SampleModel sm =
+                cm.createCompatibleSampleModel(
+                        reference.getSampleModel().getWidth(),
+                        reference.getSampleModel().getWidth());
```

Comment:
Although the code before the patch was already like this at the beginning, shouldn't be the second param 
reference.getSampleModel().getHeight()?